src/base/memory_monitor_sr.cpp

```diff
@@ -402,29 +485,116 @@ namespace cubperf
     m_mmon = mmon;
   }
 
-  int memory_monitor::aggregater::get_server_info (const MMON_SERVER_INFO &info)
+  void memory_monitor::aggregater::get_server_info (MMON_SERVER_INFO &info) const
   {
-    return 0;
+    strncpy (info.name, m_mmon->m_server_name.c_str (), m_mmon->m_server_name.size ());
+    info.total_mem_usage = m_mmon->m_total_mem_usage.load ();
+  }
+
+  int memory_monitor::aggregater::get_module_info (MMON_MODULE_INFO *&info, int module_index) const
+  {
+    int idx = 0;
+    int error = NO_ERROR;
+    if (module_index == -1)
+      {
+	// aggregate all detail memory information of modules
+	for (idx; idx < MMON_MODULE_LAST; idx++)
```

Comment:
nit.
```suggestion
	for (int idx = 0; idx < MMON_MODULE_LAST; idx++)
```

Reply:
`idx` is initialized with 0 when declared, and I intend to use this variable both for aggregating entire module information and aggregating specific modules with `module_index`.

Reply:
Why not just using 0 instead of idx for a request for one module? it's more obvious to use the constant.

Reply:
Ok. I fix it.